### ISSUE TITLE
can you provide details on what the verifier model is trained on

### ISSUE THREAD

**Issue by atinsood** [TIGER-AI-Lab/General-Reasoner#1]
thank you for elaborate paper and open sourcing the code and dataset. but can you provide some details on what dataset the verifier model is trained on. 

the paper says that you use Gemini to generated an answer and then verify along with CoT whether the the given answer is correct or not and then you use this dataset to trained the qwen 1.5 b model. 

but what is the seed dataset that contains the original question/answer pairs that the gemini model uses to generate answers against and then verify against.

**Comment by MXueguang** [comment]
Hi @atinsood, I just uploaded the dataset to train the verifier at:
[TIGER-Lab/verifier-sft-data](https://huggingface.co/datasets/TIGER-Lab/verifier-sft-data)

**Comment by atinsood** [comment]
@MXueguang can you provide more details on how this dataset was created. is it a subset of web instruct data as well, or was this generated in a different manner and what does the makeup of the data look like (does it has the same distribution as your training data in terms of maths, wikipedia etc. )

and thanks again for sharing the work.,

**Comment by haibo-qiu** [comment]
Hi @MXueguang , thank you very much for sharing your excellent work.

I have the same question as @atinsood . Could you please provide more details about how this dataset was constructed?

**Comment by MXueguang** [comment]
Hi @haibo-qiu  thanks for the questions 
and @atinsood, sorry that I missed your followup question. 

Yes, it is a subset of web instructed data, having the same distribution (and aslo overlap) with the RL training questions.
The questions for verifier training are sampled from the question set before the filtering of too easy and too hard questions.
We take the verification output from Gemini as the supervision target to train the 1.5B verifier.

**Comment by haibo-qiu** [comment]
Hi @MXueguang , thank you so much for your response. 

I have an additional question: Regarding the verifier's training data, the ground truth answers are derived from web-instructed data, and the alignment judgment along with the rationale is provided by Gemini. Could you provide more details about how the student answer is constructed? 

Looking forward to your response~

**Comment by MXueguang** [comment]
Hi @haibo-qiu, the student answer is also generated by gemini.

For a given question Q, we have extracted short answer A from web-instructed data as ground truth, and we use Gemini to generate a CoT solution end with a with predicted short answer S (student answer). Then we again use Gemini, to generate the verification rationale. Eventually, this rationale is used to train the smaller model.

**Comment by haibo-qiu** [comment]
Hi @MXueguang , that's great! 

I have no further questions. Thank you once again for your response.
